sdk/core/azure-core-experimental/src/main/java/com/azure/core/experimental/serializer/AvroSerializerProvider.java

```diff
@@ -3,19 +3,35 @@
 
 package com.azure.core.experimental.serializer;
 
-import com.azure.core.util.serializer.ObjectSerializer;
-
 /**
- * An interface to be implemented by any azure-core plugin that wishes to provide an Avro {@link ObjectSerializer}
+ * An interface to be implemented by any azure-core plugin that wishes to provide an Avro {@link AvroSerializer}
  * implementation.
  */
 public interface AvroSerializerProvider {
 
     /**
-     * Creates a new Avro-based {@link ObjectSerializer} tied to the given schema.
+     * Creates a new {@link AvroSerializer} tied to the given schema.
      *
      * @param schema The Avro schema that will be associated to the serializer.
-     * @return A new Avro-based {@link ObjectSerializer} instance.
+     * @return A new {@link AvroSerializer} instance.
+     */
+    AvroSerializer createInstance(String schema);
+
+    /**
+     * Returns the Avro schema for specified object.
+     *
+     * @param object The object having its Avro schema retrieved.
+     * @return The Avro schema for the object.
+     * @throws IllegalArgumentException If the object is an unsupported type.
+     */
+    String getSchema(Object object);
```

Comment:
Added these new methods to the provider as they are tied to the specific implementation and not a specific serializer instance.